app/services/fetch_coffees/workshop_coffee.rb

```diff
@@ -0,0 +1,54 @@
+module FetchCoffees
+  class RaveCoffee
+    require 'open-uri'
+    require 'net/http'
+    require 'json'
+    require 'nokogiri'
```

Comment:
Looking at these scrapers there's a lot repetition. Before we go any further it might be worth breaking this scrape method down and making use of instance variables. For example, the parent module could set up the imports. One method could handle the request. Another could handle the looping through objects and adding to a list. I know scrapers need to be specific but there are some parts that could be made more DRY.

Reply:
yeah, I agree in principle, the problem is that I think there is slightly too much variance.

We can definitely abstract something to make the request and get the nokgiri nodeset, but the actually getting and collating of the data I think is going to be too varied for an abstraction to make sense.